### PR TITLE
Add ellswift usage example

### PR DESCRIPTION
**Pull request by theStack**
This should hopefully be useful as orientation for users implementing the shared secret derivation part of BIP324. Conceptually the example is not very different to the ECDH one, so a lot of code/comments are just copied (e.g. context creation, secret key generation, shared secret comparison, console output, cleanup with secret key clearing).

### CODE REVIEW

**Review by real-or-random — COMMENTED** [review]
Concept ACK

File: .gitignore [hunk 1]
```diff
@@ -10,6 +10,7 @@ ctime_tests
 ecdh_example
 ecdsa_example
 schnorr_example
+ellswift_example
```
Comment (hebasto) [inline .gitignore:13]: nit: I understand that this doesn’t adhere to the surrounding code, but, strictly speaking, it should be
```suggestion
/ellswift_example
```

**Review by hebasto — COMMENTED** [review]
31f84595c4fa8fcc89021790257b85416aae44f6

I have reviewed both build systems changes and they look OK.

**Review by real-or-random — APPROVED** [review]
utACK 31f84595c4fa8fcc89021790257b85416aae44f6

File: .gitignore (hunk 1 above)
Comment (real-or-random) [reply .gitignore:13]: Yeah, feel free to adjust this for the binaries, but I guess either variant is fine (we also have the pattern `*.exe`, which may in rare cases a bit too coarse`...)

**Review by jonasnick — APPROVED** [review]
ACK 31f84595c4fa8fcc89021790257b85416aae44f6

**Pull request merged**